typing/typecore.ml

```diff
@@ -4317,7 +4324,7 @@ and type_expect_
         raise(Error(loc, env, Label_not_mutable lid.txt));
       rue {
         exp_desc = Texp_setfield(record,
-          Value_mode.regional_to_global_alloc rmode,
+          Value_mode.regional_to_local_alloc rmode,
```

Comment:
This is the BUG fix. The `alloc_mode` on `Texp_setfield` says whether the record is on heap or potentially on stack; this is passed onto later phases to maintain GC invariants. Now, if the record is `regional` it should be mapped to `local` because it might be on stack.